src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/db/RecallReasonAuditRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,123 @@
+  private val createdByUserId = ::UserId.random()
+  private val recallId = ::RecallId.random()
+  private val now = OffsetDateTime.now()
+  private val recall = Recall(recallId, nomsNumber, createdByUserId, now, FirstName("Barrie"), null, LastName("Badger"))
+  private val currentUserId = ::UserId.random()
+
+  private val underTest = RecallReasonAuditRepository(jpaRepository)
+  private val recallRepository = RecallRepository(jpaRecallRepository)
+
+  @BeforeEach
+  fun `setup createdBy user`() {
+    createUserDetails(currentUserId)
+    createUserDetails(createdByUserId)
+  }
+
+  private fun createUserDetails(userId: UserId) {
+    userDetailsRepository.save(
+      UserDetails(
+        userId,
+        FirstName("Test"),
+        LastName("User"),
+        "",
+        Email("test@user.com"),
+        PhoneNumber("09876543210"),
+        CaseworkerBand.FOUR_PLUS,
+        OffsetDateTime.now()
+      )
+    )
+  }
+
+  @Test
+  @Transactional
+  fun `can get audit info for single field updated recall`() {
+    recallRepository.save(recall, currentUserId)
+
+    assertThat(recallRepository.getByRecallId(recallId), equalTo(recall.copy(lastUpdatedByUserId = currentUserId.value)))
+
+    val recallToUpdate = recall.copy(reasonsForRecall = setOf(ELM_FURTHER_OFFENCE, ELM_EQUIPMENT_TAMPER, OTHER))
+    recallRepository.save(recallToUpdate, currentUserId)
+
+    val recallReasonAudits = underTest.recallReasonAuditForRecallId(recall.id)
+    assertThat(recallReasonAudits.size, equalTo(1))
+    val recallReasonActual = Json.decodeFromString<Array<String>>(recallReasonAudits[0].updatedValue)
+
+    assertThat(recallReasonActual.size, equalTo(3))
+    assert(recallReasonActual.contains("OTHER"))
+    assert(recallReasonActual.contains("ELM_FURTHER_OFFENCE"))
+    assert(recallReasonActual.contains("ELM_EQUIPMENT_TAMPER"))
+  }
+
+  @Test
+  @org.springframework.transaction.annotation.Transactional
+  fun `can get audit summary for updated recall`() {
+    assertThat(underTest.count(), equalTo(0))
+    recallRepository.saveAndFlush(recall.copy(lastUpdatedByUserId = currentUserId.value))
+
+    assertThat(recallRepository.getByRecallId(recallId), equalTo(recall.copy(lastUpdatedByUserId = currentUserId.value)))
+
+    val lastUpdatedDateTime1 = OffsetDateTime.now()
+    val recallToUpdate = recall.copy(lastUpdatedDateTime = lastUpdatedDateTime1, reasonsForRecall = setOf(ELM_FURTHER_OFFENCE, ELM_EQUIPMENT_TAMPER))
+    recallRepository.saveAndFlush(recallToUpdate)
+
+    val lastUpdatedDateTime2 = OffsetDateTime.now()
+    val recallToUpdate2 = recall.copy(lastUpdatedDateTime = lastUpdatedDateTime2, reasonsForRecall = setOf(ELM_FURTHER_OFFENCE))
+    recallRepository.saveAndFlush(recallToUpdate2)
+
+    assertThat(underTest.count(), equalTo(3))
+    val recallReasonAudits = underTest.auditSummaryForRecallId(recall.id)
+    assertThat(recallReasonAudits!!.columnName, equalTo("reasons_for_recall"))
+    assertThat(recallReasonAudits.auditCount, equalTo(2))
```

Comment:
Fine that count is 2 but asserts on the content would help understand the expected response.

Reply:
That is the content :) . This is the summary view. AuditCount tells the UI how many historical entries we have, so it can decide whether to show the viewHistory link or not (meant to only show link where auditCount > 1)